Objects/object.c

```diff
@@ -2933,7 +2935,9 @@ _PyTrash_thread_destroy_chain(PyThreadState *tstate)
         op->ob_tid = 0;
         _Py_atomic_store_ssize_relaxed(&op->ob_ref_shared, _Py_REF_MERGED);
 #else
-        tstate->delete_later = (PyObject*) _PyGCHead_PREV(_Py_AS_GC(op));
+        uintptr_t refcnt = *((uintptr_t*)op);
```

Comment:
The `-1` operation is a bit magic, I suggest adding this comment:

```suggestion
        /* Get the delete_later pointer from the refcnt field.
         * See _PyTrash_thread_deposit_object(). */
        uintptr_t refcnt = *((uintptr_t*)op);
```